contentcuration/contentcuration/static/js/administration/router.js

```diff
@@ -0,0 +1,293 @@
+		selected: true,
+	},
+	id: {
+		label: "Channel ID",
+	},
+	priority: {
+		label: "Priority",
+	},
+	users: {
+		label: "# of Users",
+	},
+	items: {
+		label: "# of Items",
+	},
+	created: {
+		label: "Date Created",
+	}
+}
+
+const USER_FILTERS = {
+	all: {
+		label: "All",
+		queryParams: {},
+		selected: true,
+	},
+	activated: {
+		label: "Active",
+		queryParams: { is_active: "True" },
+	},
+	not_activated: {
+		label: "Inactive",
+		queryParams: { is_active: "False" },
+	}, 
+	admins: {
+		label: "Admins",
+		queryParams: { is_admin: "True" },
+	},
+	is_chef: {
+		label: "Sushi Chefs",
+		queryParams: { ricecooker_version: ">0.0.0" },
+	}
+}
+
+const USER_SORT_FILTERS = {
+	email: {
+		label: "Email",
+	},
+	first_name: {
+		label: "First Name",
+	},
+	last_name: {
+		label: "Last Name",
+	},
+	date_joined: {
+		label: "Date Joined",
+	},
+	channels: {
+		label: "# of Channels",
+	},
+}
+
+const SORT_ORDER_OPTIONS = {
+	ascending: {
+		label: "Ascending",
+		selected: true,
+	},
+	descending: {
+		label: "Descending",
+	},
+}
+
+var AdministrationRouter = Backbone.Router.extend({
+
+    routes: {
+		"": "default",
+		"/": "default",
+		"users/": "users",
+		"channels/": "channels",
+        "users(/filter/:filter)(/sort/:key-:order)(/search/:search)(/p:page)(/:pagesize-per-page)":        "users",    // #users
+        "channels(/filter/:filter)(/sort/:key-:order)(/search/:search)(/p:page)(/:pagesize-per-page)":     "channels",    // #channels
+    },
+	execute(callback, args, name){
+		let page = args[4] ? Number(args[4]) : 1
+		let newRouteParams = {
+			name: name,
+			filter: args[0],
+			sortKey: args[1],
+			sortOrder: args[2],
+			search: args[3],
+			page: page,
+			pageSize: Number(args[5]),
+		}
+		let sameTabNewParams = this.currentRouteParams.name === newRouteParams.name &&
+								!_.isEqual(this.currentRouteParams, newRouteParams)
+
+		let switchTabNewParams = this.currentRouteParams.name !== newRouteParams.name &&
+								!_.isEqual(this.routeParamsCache[newRouteParams.name], newRouteParams)
+		if ( 
+			// check to see if we need to reload
+			sameTabNewParams || switchTabNewParams
+		){
+			// if so, execute the route
+			if (callback) callback.apply(this, args);
+		}
+
+		// update the current state
+		this.currentRouteParams = newRouteParams
+		this.routeParamsCache[name] = this.currentRouteParams
+
+		// make sure the correct tab is showing, even if we aren't loading anything
+		$('a.btn.'+name).tab('show')
+		return false
+	},
+	getRoute({name, filter, sortKey, sortOrder, search, page, pageSize}){
+		return name +
+				(filter ? "/filter/" + filter : "") +
+				(	
+					(sortKey || sortOrder) ? 
+					"/sort/" + (sortKey ? sortKey : GET_DEFAULT_ROUTES()[name]['sortKey']) + "-" +
+					(sortOrder ? sortOrder : "ascending")
+					: ""
+				) +
+				(search ? "/search/" + search : "") +
+				(page ? "/p" + page : "") +
+				(pageSize ? "/" + pageSize + "-per-page" : "")
+	},
+	gotoRouteForParams(params){
```

Comment:
This function takes incoming parameters, merges them with existing router state, then navigates to that route.